osmaxx/core/templatetags/navigation.py

```diff
@@ -1,19 +1,34 @@
-# Adopted from https://djangosnippets.org/snippets/2783/
+from urllib.parse import urlsplit, urlunsplit
+
+import logging
 from django import template
 from django.conf import settings
+from django.http import HttpRequest
+
+logger = logging.getLogger(__name__)
 
 register = template.Library()
 
 
 @register.filter
-def siteabsoluteurl(absoluteurl, request):
+def siteabsoluteurl(relative_or_absolute_url: str, request: HttpRequest) -> str:
```

Comment:
IMO unnecessary addition, because this function is used in templates only. It doesn't help to understand the code, the parameters are already quite sensibly named without the need of type information. Furthermore we don't use that at any other place in the project, unless we plan to include this pattern in all place in the future, I think we should not start using it.